### ISSUE TITLE
Can you add anti-retraction?

### ISSUE THREAD

**Issue by llm2580** [risin42/NagramX#10]
This can be useful in some cases, and it also provides permanent access to photos that should have been destroyed, which is shameless, but I still want this feature.So I really want you to add this feature.thank you so much.

**Comment by risin42** [comment]
It's an interesting idea and I might look into it at some point, but adding new features that violate the TOS isn't going to be a high priority for me.

**Comment by Qing0731** [comment]
@risin42 It seems that there is a bug in the function of saving edited messages. When someone sends an APK file (but not an APK image), and then the text content is edited, an incorrect empty record will appear.
![Screenshot_2025-01-09-22-54-59-63_7d47d81692b309eee41621f70e9c1da1.jpg](https://github.com/user-attachments/assets/5261fea0-2751-4ea0-ade7-eee854c7146f)

**Comment by risin42** [comment]
Actually, that's not a bug. I disabled the part that saves media from deleted messages in public chats because it might hog your storage. You can see if the same thing happens in private chats. I'm planning to add a setting so you can choose if you want to save media in public chats or not.

**Comment by Qing0731** [comment]
Indeed, this is not the case in private chat, looking forward to you joining this new feature

**Comment by Qing0731** [comment]
In private chats, it seems that blocking deletion doesn't work very well

**Comment by risin42** [comment]
Could you please open a new issue for this with more details? Thanks.

**Comment by Qing0731** [comment]
OK

**Comment by risin42** [comment]
> Actually, that's not a bug. I disabled the part that saves media from deleted messages in public chats because it might hog your storage. You can see if the same thing happens in private chats. I'm planning to add a setting so you can choose if you want to save media in public chats or not.

5328eb37e44eaeac751871affc83a61fb8a7bfce
